src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -361,35 +362,70 @@ export class CommandDetectionCapability extends Disposable implements ICommandDe
 		this._logService.debug('CommandDetectionCapability#handleCommandStart', this._currentCommand.commandStartX, this._currentCommand.commandStartMarker?.line);
 	}
 
-	private _handleCommandStartWindows(): void {
+	private async _handleCommandStartWindows(): Promise<void> {
 		this._currentCommand.commandStartX = this._terminal.buffer.active.cursorX;
 
 		// On Windows track all cursor movements after the command start sequence
 		this._commandMarkers.length = 0;
-		// HACK: Fire command started on the following frame on Windows to allow the cursor
-		// position to update as conpty often prints the sequence on a different line to the
-		// actual line the command started on.
-		timeout(0).then(() => {
-			if (!this._currentCommand.commandExecutedMarker) {
-				this._onCursorMoveListener = this._terminal.onCursorMove(() => {
-					if (this._commandMarkers.length === 0 || this._commandMarkers[this._commandMarkers.length - 1].line !== this._terminal.buffer.active.cursorY) {
-						const marker = this._terminal.registerMarker(0);
-						if (marker) {
-							this._commandMarkers.push(marker);
-						}
-					}
-				});
+
+		const lastCommand = this.commands.at(-1);
+		const terminal = this._terminal;
+
+		function cursorOnCorrectLine(): boolean {
```

Comment:
Name could be improved "correct" is ambiguous

Reply:
Yeah I wondered about that too. `Next` is relative, so isn't great either.

Reply:
`isCursorLineWithLastCommand`?